File: examples/add_job.rs
use bullrs::queue::Queue;
use bullrs::job::Job;
use uuid::Uuid;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    println!("bullrs main: add job");

    let queue = Queue::new("test-queue", "redis://localhost:6379").await?;

    let uuid = Uuid::new_v4();

    println!("uuid: {}", uuid);

    let job = Job::new(&format!(r#"{{"uuid":"{}"}}"#, uuid));

    queue.add_job(job).await?;

    Ok(())
}

File: nodejs/add_job.js
const Queue = require('bull')
const crypto = require('crypto')

const testQueue = new Queue('test-queue', 'redis://localhost:26379')

;(async () => {
  const uuid = crypto.randomUUID()
  await testQueue.add({ uuid })

  console.log({ uuid })

  process.exit()
})()
